src/main/scala/io/lambdaworks/workshop/exceptions/PasswordChecker.scala

```diff
@@ -2,15 +2,47 @@ package io.lambdaworks.workshop.exceptions
 
 object PasswordChecker {
 
-  def validate(password: String): Either[List[Throwable], String] = ???
-
-  private def minNumberOfChars(password: String, length: Int): Either[Throwable, String] = ???
-
-  private def containsUpperCase(password: String): Either[Throwable, String] = ???
-
-  private def containsLowerCase(password: String): Either[Throwable, String] = ???
-
-  private def containsNumber(password: String): Either[Throwable, String] = ???
+  def validate(password: String): Either[List[Throwable], String] = {
+    val passwordCheck1 = minNumberOfChars(password, 5)
+    val passwordCheck2 = containsUpperCase(password)
+    val passwordCheck3 = containsLowerCase(password)
+    val passwordCheck4 = containsNumber(password)
```

Comment:
Maybe, you can think about some more meaningful names here. Instead of `passwordCheck1` it can be `meetMinLength` for example.